sonar-plugin-api/src/main/java/org/sonar/api/scan/filesystem/PathResolver.java

```diff
@@ -98,6 +105,10 @@ private static File parentDir(Collection<File> dirs, File cursor) {
     return null;
   }
 
+  /**
+   * @deprecated since 6.0 was used when component keys were relative to source dirs
+   */
+  @Deprecated
   public static final class RelativePath {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)